### ISSUE TITLE
Make tests/shtest portable to Windows

### ISSUE THREAD

**Issue by nicowilliams** [stedolan/jq#1079]
It runs, but fails early.  For example, there's no /dev/urandom on Windows.

**Comment by JanSchulz** [comment]
for test failures, I push a commit to my repo where I enabled all tests again: https://ci.appveyor.com/project/JanSchulz/jq/build/1.0.41

**Comment by nicowilliams** [comment]
Yeah, and this is the log from my build prior to disabling the failing tests: https://ci.appveyor.com/project/nicowilliams/jq/build/1.0.1/job/fsxq3h06tt8oq448

We'll also need to add regression tests for jq bugs on Windows.

**Comment by nicowilliams** [comment]
The first problem in `tests/shtest`, of course, is that the warnings we're looking for have Windows line endings (`\r\n`).
